pkg/chaosdaemon/tasks/task_manager.go

```diff
@@ -0,0 +1,278 @@
+// Copyright 2021 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package tasks
+
+import (
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+
+	"github.com/chaos-mesh/chaos-mesh/pkg/chaoserr"
+)
+
+var ErrCanNotAdd = errors.New("can not add")
+var ErrCanNotAssign = errors.New("can not assign")
+
+// Injectable stand for the base behavior of task : inject a process with PID.
+type Injectable interface {
+	Inject(pid PID) error
+}
+
+// Recoverable introduce the task recovering ability.
+// Used in Recover.
+type Recoverable interface {
+	Recover(pid PID) error
+}
+
+// Creator init an Injectable with values.
+// We use it in a case that TaskConfig.data init an Injectable task here.
+type Creator interface {
+	New(values interface{}) (Injectable, error)
+}
+
+// Assign change some of an Injectable task with its own values.
+// We use it in a case that we use TaskConfig.data
+// to update an Injectable task.
+type Assign interface {
+	Assign(Injectable) error
+}
+
+// TaskExecutor indicate that the type can be used
+// for execute task here as a task config.
+// Addable means we can sum many task config in to one for apply.
+// Creator means we can use the task config to create a running task
+// which can Inject on PID.
+// Assign means we can use the task config to update an existing running task.
+type TaskExecutor interface {
+	Addable
```

Comment:
Does it mean that we will migrate `PodIOChaos` into the task manager in the future?

Reply:
I will do it in future , if you think it is OK. And I'm glad to do it.